uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/logging/FileLogger.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    @Override
+    public void w(@NonNull String tag, @NonNull String message) {
+        log(LogLevel.WARN, tag, message, null);
+    }
+
+    @Override
+    public void w(@NonNull String tag, @NonNull String message, @NonNull Throwable exception) {
+        log(LogLevel.WARN, tag, message, exception);
+    }
+
+    @Override
+    public void i(@NonNull String tag, @NonNull String message) {
+        log(LogLevel.INFO, tag, message, null);
+    }
+
+    @Override
+    public void i(@NonNull String tag, @NonNull String message, @NonNull Throwable exception) {
+        log(LogLevel.INFO, tag, message, exception);
+    }
+
+    @Override
+    public void v(@NonNull String tag, @NonNull String message) {
+        log(LogLevel.VERBOSE, tag, message, null);
+    }
+
+    @Override
+    public void v(@NonNull String tag, @NonNull String message, @NonNull Throwable exception) {
+        log(LogLevel.VERBOSE, tag, message, exception);
+    }
+
+    private void log(LogLevel logLevel, String tag, String message, Throwable throwable) {
+        final String logMessage = formatMessage(logLevel, tag, message, throwable);
+        writeLogToFile(logMessage);
+    }
+
+    private void writeLogToFile(final String logMessage) {
+        try {
+            final File logFile = getLogFile();
+            //BufferedWriter for performance, true to set append to file flag
+            BufferedWriter buf = new BufferedWriter(new FileWriter(logFile, true));
+            buf.append(logMessage);
+            buf.newLine();
+            buf.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public File getLogFile() throws IOException {
+        final Context context = ApplicationProvider.getApplicationContext();
+        final File directory = context.getFilesDir();
+        final File logFile = new File(directory, mFileName);
+
+        if (!logFile.exists()) {
+            final boolean fileCreated = logFile.createNewFile();
+            if (!fileCreated) {
+                throw new IOException("Unable to create new log file :(");
+            }
+        }
+
+        return logFile;
+    }
+
+    public String getLogFileName() {
+        return mFileName;
+    }
+
+    private String formatMessage(
+            @NonNull final LogLevel logLevel,
+            @NonNull final String tag,
+            @NonNull final String message,
+            @Nullable final Throwable throwable) {
+        final String logMessage = getUTCDateTimeAsString() + ": " + logLevel.getLabel() + "/" + tag
+                + ": " + message;
+        return logMessage + (throwable == null ? "" : '\n' + Log.getStackTraceString(throwable));
+    }
+
+    private static String getUTCDateTimeAsString() {
+        final SimpleDateFormat dateFormat = new SimpleDateFormat(DATE_FORMAT, Locale.getDefault());
+        dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+        return dateFormat.format(new Date());
```

Comment:
Can we, conditionally, on API > 26, use the java.time methods for this rather than SimpleDateFormat?  

Something like:

// Alternately, use ISO_INSTANT here.
return DateTimeFormatter .RFC_1123_DATE_TIME.format(Instant.now());